### ISSUE TITLE
[VO + Safari] Negatives only read once before needing to tab away and back

### ISSUE THREAD

**Issue by phet-steele** [phetsims/john-travoltage#238]
@jessegreenberg the first time I tab to the arm or leg, I hear the negative read out loud. This is assuming that the appendage is already at a negative position before I tab to it since further announcements do not read the negative after using the arrow keys. I must tab away and back to an appendage to hear a negative again, but then I hear no further negatives until I repeat this process.

Seen on macOS 10.12.4 Safari. For phetsims/tasks/issues/831

**Comment by jessegreenberg** [comment]
Thanks @phet-steele.  I can't reproduce this, but I am on 10.12.5 beta, that might be the reason. I hear:

Position: minus 4, hand pointing at upper door, close to doorknob [on focus]
Position: minus sign 5, away from doorknob
Position: minus sign 6, not so close to doorknob

...and so on.

**Comment by jessegreenberg** [comment]
Over skype @phet-steele mentioned that this was no longer happening in 10.12.4 Safari.

VO might have been caching an old version of john-travoltage_en.html from spot. Or maybe it was just not reading the unicode minus sign for another buggy reason.  If it comes up again, we can investigate further.

**Comment by jessegreenberg** [comment]
@phet-steele corrected me, saying it was quickly broken again :( I will see if I can reproduce the problem.

**Comment by jessegreenberg** [comment]
This is a problem with VoiceOver's "Karen" voice.  When I switched to that voice (the one @phet-steele was using to test), this started to happen consistently.  Otherwise, I could not reproduce this.

**Comment by jessegreenberg** [comment]
"Karen" is an Australian voice, it might have something to do with that.

**Comment by jessegreenberg** [comment]
When I switch to "hyphe-minus" unicode symbol, I get the exact opposite issue.  On focus, I hear no minus sign. Every subsequent movement I *do* hear "Minus [number]".

**Comment by jessegreenberg** [comment]
Using a combination of the two unicode values depending on whether or not the appendage has focus seems to fix this.

**Comment by jessegreenberg** [comment]
But then the default voice doesn't read out the minus sign when the value changes.

**Comment by jessegreenberg** [comment]
'\u002D' - "hyphen-minus", not read by default VoiceOver voice
'\u2212' - "Minus Sign", not read when ```aria-valuetext``` changes by Australian VoiceOver voice (and maybe others).

@terracoda @jobara do you know of any way to get around this?

**Comment by jessegreenberg** [comment]
@emily-phet is this worth more time?

**Comment by jessegreenberg** [comment]
I will try a few other english voices.

**Comment by emily-phet** [comment]
@jessegreenberg If this is a problem just with the "Karen" voice, then we should not fix this.

**Comment by jessegreenberg** [comment]
Ok, thanks @emily-phet, I am going to try these voices, @phet-steele is going to try a few others.

- Veena (India)
- Moira (Ireland)
- Fiona (Scottland)

**Comment by jessegreenberg** [comment]
All of the voices in https://github.com/phetsims/john-travoltage/issues/238#issuecomment-299514066 have this problem.

**Comment by phet-steele** [comment]
The voices I tried both have the problem:

- Oliver (English, UK)
- Angelica (Spanish, Spain)

**Comment by jessegreenberg** [comment]
"Samantha"  (US) has this problem.

**Comment by jessegreenberg** [comment]
Are "Alex" (US, default voice) and "Fred" (US, not default) the only voices that don't do this?

**Comment by jessegreenberg** [comment]
One thing we could to is add the word "minus" in front of the value.  It won't look pretty in the PDOM, but it will always sound correct.

**Comment by jessegreenberg** [comment]
I don't know much about unicode values or why certain voices would ignore them. I am unclear if this is a VoiceOver feature or a bug. Without knowing more about locales and unicode values it seems like a bug.

@emily-phet @terracoda @jobara @jhung is https://github.com/phetsims/john-travoltage/issues/238#issuecomment-299516563 an acceptable workaround?

It would look like this:
<img width="725" alt="screen shot 2017-05-05 at 10 59 24 am" src="https://cloud.githubusercontent.com/assets/6396244/25755903/fe46a880-3181-11e7-9463-7d9d8424c482.png">

**Comment by emily-phet** [comment]
@jessegreenberg This is so odd - my understanding is that the point of Unicode is to avoid exactly this situation. I'm trying to find more information about how the different voices differ (in a technical sense) but I'm not finding much that would be helpful. One thought is that each voice is accessing different (or customized) libraries to recognize symbols, and perhaps some of the dialects don't include this symbol, but I can't imagine why it would be structured that way. 

Is there a way to check if other occurrences of the minus Unicode run into this same issue? So what if you just have a screen reader only the Unicode in multiple voices or only the Unicode with a number before or after, do they have the same problem out of context? I'm just wondering if there is something about the context in which we have the minus is somehow causing different voices to interpret the information differently. Not sure why that would be, but I can't think of any other possible issues beyond it just being a really weird bug.

**Comment by jessegreenberg** [comment]
@emily-phet context does seem important, based on https://github.com/phetsims/john-travoltage/issues/238#issuecomment-299336713 and https://github.com/phetsims/john-travoltage/issues/238#issuecomment-299339259.  We can quickly make a simple test case with some different contexts using the unicode symbol.

**Comment by jessegreenberg** [comment]
Here is an example JSFiddle: https://jsfiddle.net/s8m2spxc/

Using voice Samantha (US, female), the only unicode that is read is "Minus Sign" ('\u2212').
Using voice Alex (US, male), the only unicode value that is read is "Minus Sign" ('\u2212').
Using voice Karen (Australian, female), the only unicode that is read is "Minus Sign" ('\u2212').

**Comment by jessegreenberg** [comment]
When using with ```aria-valuetext``` things start to get weird.
https://jsfiddle.net/s8m2spxc/2/

Testing with "Karen" - 
When focus lands on the first slider (using '\u2212' - "Minus Sign"), "Karen" reads the "Minus sign". When changing the value on that slider, the minus sign is never read.

When focus lands on the second slider (using '\u002D' - "hyphen-minus"), "Karen" does *not* read "Minus Sign", but when the value changes, the minus sign *is* read. Which is exactly what is happening in john-travoltage as well.

**Comment by jessegreenberg** [comment]
When using voice "Alex", we have the exact same behavior as above, but when changing the value with  '\u2212' - "Minus Sign" in the ```aria-valuetext```, the unicode value "Minus Sign" is read.

@emily-phet I think the above JSFiddle isolates this as a VoiceOver bug.  Should we mark this as wontfix and submit a bug report?

**Comment by jessegreenberg** [comment]
Or should we add "minus" in front of the number like https://github.com/phetsims/john-travoltage/issues/238#issuecomment-299519447?

**Comment by emily-phet** [comment]
@jessegreenberg Let's submit a bug report, and also add "negative" in front of the negative number (removing the "-" sign). I hate to have to do that, but I think there will be a significant number of VO users, and it will make learning more difficult to not hear (or inconsistently hear) the "negative" position indicator.

**Comment by jessegreenberg** [comment]
Sounds good @emily-phet.  The workaround was added in the above commit.  Marking this as wontfix and leaving this issue open until we hear from Apple when we submit a bug report in https://github.com/phetsims/a11y-research/issues/37.

**Comment by terracoda** [comment]
@emily-phet, sorry I was not available for earlier comment on this issue. If this continues to be an issue, we might want to use words to indicate above and below the door knob.

**Comment by emily-phet** [comment]
Any word from Apple? Can this be closed?

**Comment by jessegreenberg** [comment]
1/8/18: Just checked the bug report, no response from Apple, and I updated phetsims/a11y-research#37. Since this impacts *all* usages of aria-valuetext with negative values (not just the ones in this sim) I think this issue can be closed and it can be tracked in phetsims/a11y-research#37.
